packages/web/server/utils/embeddings.ts

```diff
@@ -136,7 +136,7 @@ function preprocessText(text: string): string {
     .trim()
 }
 
-function chunkIssue(issue: Issue | RestIssue, exclude?: Set<string>) {
+export function chunkIssue(issue: Pick<Issue | RestIssue, 'labels' | 'title' | 'body'>, exclude?: Set<string>) {
```

Comment:
NIT: Fairly opposed to string literals as of TS5.8, would suggest instead opting for a mapped type w/as const. The trade-off of less flexibility and a little more TS syntactical complexity here is worthwhile when gaining the stronger type safety and type inference when using an indexed access type. And I'd add that personally I find `Pick` more readable with just types in it and no strings.
```suggestion
export const issueSegments = { labels: 'labels', title: 'title', body: 'body', } as const;

export type IssueSegments = (typeof issueSegments)[keyof typeof issueSegments];

export function chunkIssue( issue: Pick<Issue | RestIssue, IssueSegments>, exclude?: Set<string>, ) { const labels = getLabels(issue).filter((l) => !exclude?.has(l)); return preprocessText( ${issue[issueSegments.title]}\n${labels.join(', ')}\n${ issue[issueSegments.body] }, ); }
```